js/novicell-overlay.js

```diff
@@ -3,266 +3,302 @@
-          // Remove elements
-          this.overlayElem.parentElement.removeChild(this.overlayElem);
-          this.backdrop.parentElement.removeChild(this.backdrop);
-  
-          // Stop listening for close overlay events
-          document.removeEventListener('keyup', () => {
-            this.destroy();
-          });
-  
-          // Remove class on body
-          document.documentElement.classList.remove(
-            'no-scroll',
-            'novi-overlay--open'
+      } else if (
+      // Check if content comes from a HTML element
+        Object.prototype.hasOwnProperty.call(this, 'element')
+                && this.element !== null
+      ) {
+        // let element = this.element;
+
+        if (this.element) {
+          this.content = this.element.innerHTML;
+
+          this.constructOverlay();
+        } else {
+          console.warn(
+            'novicell.overlay: element does not exist. Please provide a valid DOM element.'
           );
-  
-          // Reset video letiable
-          this.isVideo = false;
-  
-          // Call this.onDestroy callback
-          if (typeof this.onDestroy === 'function') {
-            this.onDestroy();
-          }
         }
-      };
-  
-      this.constructOverlay = function constructOverlay() {
-        // Create backdrop
-        this.setupBackdrop();
-  
-        // Create the overlay
-        this.setupOverlay();
-  
-        // Create content for overlay
-        this.setupOverlayContainer();
-  
-        // Create close button
-        this.setupCloseButton();
-  
-        // Add class to body-element
-        document.documentElement.classList.add('no-scroll');
-  
-        // Call this.onLoaded callback
-        if (typeof this.onLoaded === 'function') {
-          this.onLoaded();
+      } else if (this.videoId) {
+        // Or if content comes from an ID
+        let src = '';
+        this.isVideo = true;
+        if (this.videoType === 'vimeo') {
+          src = `https://player.vimeo.com/video/${this.videoId}?autoplay=${this.autoplay}&loop=1&muted=1`;
+        } else if (this.videoType === 'youtube') {
+          src = `https://www.youtube.com/embed/${this.videoId}?autoplay=${this.autoplay}&rel=0`;
+        } else {
+          return;
         }
-      };
-  
-      this.setupBackdrop = function setupBackdrop() {
-        // Create the backdrop
-        this.backdrop = document.createElement('div');
-        this.backdrop.classList.add('novi-backdrop');
-        this.backdrop.id = 'js-novi-backdrop';
-  
-        this.backdrop.addEventListener('click', (e) => {
-          if (
-            e.target.classList.contains('novi-overlay')
-                      || e.target.classList.contains('novi-overlay__container')
-          ) {
-            this.destroy();
-          }
+        const iframe = document.createElement('iframe');
+        iframe.setAttribute('src', src);
+        iframe.setAttribute('frameborder', 0);
+        iframe.setAttribute('allow', 'autoplay');
```

Comment:
should we also add `fullscreen` to the allow-attribute, as in Vimeo embedcode?

Reply:
The attribute allowfullscreen is soon deprecated, and we should ideally just use the "allow" attribute instead. 

So yes, we need to do like in vimeo, both have 
allow="autoplay; fullscreen"
and
allowfullscreen 
or it won't work in edge or firefox